src/shared/features/AppletSettings/ReportConfigSetting/ReportConfigSetting.hooks.ts

```diff
@@ -0,0 +1,48 @@
+import { useParams } from 'react-router-dom';
+
+import { storage } from 'shared/utils';
+
+import { verifyReportServer, setPasswordReportServer } from './ReportConfigSetting.utils';
+import { UseCheckReportServer } from './ReportConfigSetting.types';
+import { OK_MESSAGE, SUCCESS_MESSAGE } from './ReportConfigSetting.const';
+
+export const useCheckReportServer = ({ url, publicKey }: UseCheckReportServer) => {
+  const { appletId = '', ownerId = '' } = useParams();
+
+  const onVerify = async () => {
+    const token = storage.getItem('accessToken');
+
+    try {
+      const response = await verifyReportServer({ url, publicKey, token: `${token}` });
+      const { message } = (await response.json()) ?? {};
+
+      return message === OK_MESSAGE;
+    } catch {
+      return false;
+    }
+  };
+
+  const onSetPassword = async (password: unknown) => {
```

Comment:
should not `password` cast only to `string` type?

Reply:
fixed